pkg/installmanager/installmanager.go

```diff
@@ -1803,10 +1807,93 @@ func (m *InstallManager) overrideCredentialsModeOnInstallConfig(credentialsMode
 	}
 
 	if err := ioutil.WriteFile(clusterConfigPath, clusterConfigBytes, 0644); err != nil {
-		return errors.Wrap(err, "error writing install-config.yaml")
+		return errors.Wrap(err, "error writing cluster-config.yaml")
 	}
 
 	m.log.WithField("newMode", credentialsMode).Info("overrode effective in-cluster credentials mode")
 
+	if cd.Spec.Platform.Azure == nil {
+		return nil
+	}
+
+	//If Azure, need fill up azure_resource_prefix and azure_resourcegroup for 99_cloud-creds-secret.yaml
+	m.log.Info("patch 99_cloud-creds-secret.yaml when platform is azure")
+	region := installConfig.Platform.Azure.Region
+
+	//Read infrastructureName/resourceGroupName from manifests cluster-infrastructure-02-config.yml
+	clusterInfraConfigPath := filepath.Join(m.WorkDir, "manifests", clusterInfraConfigYAML)
+	clusterInfraConfigBytes, err := ioutil.ReadFile(clusterInfraConfigPath)
+	if err != nil {
+		return errors.Wrap(err, "error reading manifests cluster-infrastructure-02-config.yml")
+	}
+
+	clusterInfraConfigJson, err := yaml.YAMLToJSONStrict(clusterInfraConfigBytes)
+	if err != nil {
+		return errors.Wrap(err, "error converting cluster-infrastructure-02-config.yml to json")
+	}
+
+	infraName := gjson.Get(string(clusterInfraConfigJson), `status.infrastructureName`).String()
+	resourceGroupName := gjson.Get(string(clusterInfraConfigJson), `status.platformStatus.azure.resourceGroupName`).String()
+	if region == "" || infraName == "" || resourceGroupName == "" {
+		return fmt.Errorf("error reading from manifests, region=%s, infrastructureName=%s, resourceGroupName=%s", region, infraName, resourceGroupName)
```

Comment:
minor point: perhaps we should include the whole `clusterInfraConfigJson` in the error message as well, as it would probably make it a lot easier to figure out what the problem was. (Could be field missing, misspelled, wrong path, ...)

Reply:
Done